### DISCUSSION TITLE
Database의 Actor화에 관련된 질문에 대한 교수님 답변

### DISCUSSION THREAD

**Discussion by Topslug** [cau-swe/checkpoint-1#32] [General]
교수님께 드렸던 질문을 여기 공유드립니다.

Q:
Database의 Actor화?에 대해서 궁금합니다.
Use Case 5 : Inspect Access History의 경우를 보면 participating Actor로서 Database가 있는데요,
그런데 궁금한 점은 Log in에 대한 정보도 Database에 있을 수 있지 않은가?입니다.
저는 그런 생각이 들어서 Log in의 경우에도 Database로 표현을 할 수 있다고 생각을 했는데, Use Case 9 : ManagementUsers의 경우에는 Database를 Actor에 포함 시키지 않았습니다!
혹시 이 둘의 차이는 무엇인가요? Log in의 경우는 로컬 컴퓨터의 Database에 있는 경우라서 사용하지 않은 것인가요?
Database의 경우 언제 Actor로 표시를 해주고 안 해주는지도 궁금한 것 같습니다!

A:
Database의 경우 Log in을 위한 데이터베이스는 시스템 내부에 있다고 보는 것이 타당합니다. (추후에 로그인 관리자를 외부에 두거나 SSO 기능으로 확장한다면 이들이 Actor가 될 수 있습니다.) 한편 Access history를 위한 데이터베이스는 시스템 내부 혹은 외부 어느 곳에 있어도 되며 예시한 UC에서는 외부에 있는 경우로 가정한 것입니다.

**Comment by Topslug** [comment]
이러한 점을 볼 때, 화상 회의 때 얘기가 나왔었던 Database의 문제는 시스템을 어떻게 구축하느냐에 따라 달린 문제인 것 같습니다.

그때 저는 Database를 Actor로 보는 것이 맞지 않냐고 질문을 드렸고, 경준님이 동의해 주셨었는데, 창선님께서 의문을 가지셨던 기억이 있습니다.
그래서 저는 시스템의 스펙에 대해 짚을 필요가 있다고 생각합니다.

그런데 이러한 시스템 스펙은 매장의 상황에 따라 달라질 수 있지 않을까 싶습니다.
만약 가맹점인 경우, 여러 개의 매장이 같은 메뉴를 공유하게 되므로 외부 데이터베이스로 보는 것이 타당한 것 같은데,
단일 매장에 서비스하는 프로그램의 경우, 매장 pc를 서버로 볼 수 있지 않은가, 그래서 시스템 내부로 볼 수 있지 않은가라는 생각이 들었습니다.

주문의 경우 매장이 여러 개인 경우 매장마다 현재 주문 목록이 만들어져야 될 것 같은데, 시스템 상에서 주문을 받았을 때, 가까운 주소의 매장을 찾아서 주문을 발송하는 과정이 필요하지 않은가 싶은 생각이 듭니다.(생각해보니 고객쪽에서 매장을 선택하는 군요, 그래도 시스템이 해당 매장에 발송한다는 얘기는 없었던 것 같은데, 참고하시면 될 것 같습니다.)
반면 단일 매장으로 가정한다면 바로 매장 쪽으로 보내기만 하면 되지 않을까 싶습니다.
이 부분에 대해서 같이 토의해보면 좋을 것 같습니다.(Problem Structure)

그래서 저는 자세히는 잘 모르겠지만, 단일 매장으로 하면 조금 상황이 간편해지지 않을까라는 생각이 듭니다.
이 부분에 대해서 같이 토의해보면 좋을 것 같습니다.

**Reply by TedLeem** [reply]
단일 매장 찬성합니다...ㅎ
